### DISCUSSION TITLE
Upgrade fails partially due to error

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#16627] [How to]
tried to upgrade 6.3 to 6.4 multiple time and cannot get following:
> create index: w_yf_servers_api_key_idx ... | Error(8) [SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes

Does existing API key is really too long?

**Comment by rskrzypczak** [comment]
You have a wrong database configuration.
This is probably innodb_default_row_format.
https://github.com/YetiForceCompany/YetiForceCRM/tree/developer/tests/setup/db
https://yetiforce.com/pl/component/zoo/item/web-server-requirements
Check your configuration in the panel: `Software configuration/Logs/Server configuration`

**Reply by s2b-git** [reply]
10.1.48-MariaDB-0ubuntu0.18.04.1

Below adjusted to upgrade scenario:


##############################################
#  Best configuration for YetiForceCRM       #
#  Created by m.krzaczkowski@yetiforce.com   #
##############################################
[client]
default-character-set 			= utf8

[mysql]
default-character-set 			= utf8

[mysqld]
user                  	 		= mysql
pid-file              	  		= /run/mysqld/mysqld.pid
socket                	  		= /run/mysqld/mysqld.sock
#port                	  		= 3306
basedir               	 		= /usr
datadir							= /var/lib/mysql
tmpdir							= /tmp
lc-messages-dir       	  		= /usr/share/mysql
bind-address					= 0.0.0.0
log_error 						= /var/log/mysql/error.log
expire_logs_days       	 		= 10

default_storage_engine			= InnoDB
character-set-server 			= utf8
collation-server     			= utf8_general_ci
init-connect 					= 'SET NAMES utf8'
sql-mode						= "ONLY_FULL_GROUP_BY,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION"
wait_timeout					= 3600
connect_timeout					= 3600
net_read_timeout				= 3600
net_write_timeout				= 3600
interactive_timeout				= 600
max_allowed_packet				= 128M
innodb_lock_wait_timeout		= 3600
innodb_file_per_table 			= ON
innodb_ft_min_token_size		= 2
innodb_stats_on_metadata		= OFF
ft_min_word_len 				= 2
innodb_default_row_format		= 'dynamic'
table_open_cache				= 1000
table_definition_cache			= 1400


# key_buffer_size 				= 500M
# key_buffer 					= 500M
# myisam_sort_buffer_size 		= 500M
# tmp_table_size				= 2G
# max_heap_table_size			= 2G
# max_connections				= 200
# innodb_buffer_pool_size 		= 2G
# innodb_buffer_pool_instances 	= 8 #(or 1 if innodb_buffer_pool_size < 1GB)
# innodb_additional_mem_pool_size = 1024M
# innodb_log_file_size 			= 512M
# innodb_log_buffer_size 		= 256M

**Reply by s2b-git** [reply]
server config has no red marks except public_html and headers - but they irrelevant I guess

**Reply by rskrzypczak** [reply]
Attach a screenshot of your CRM configuration.
![image](https://user-images.githubusercontent.com/10221999/199916107-c3b0ca65-19da-4292-b718-d12a2489d62f.png)
